mlflow/pyfunc/__init__.py

```diff
@@ -366,11 +373,13 @@ def all_ints(xs):
 
         raise MlflowException(
             "Incompatible input types for column {0}. "
-            "Can not safely convert {1} to {2}.{3}".format(name, values.dtype, numpy_type, hint)
+            "Can not safely convert {1} to {2}.{3}".format(
+                name, values.dtype, numpy_type, hint
+            )
         )
 
 
-def _enforce_schema(pdf: PyFuncInput, input_schema: Schema):
```

Comment:
@tomasatdatabricks Are `PyFuncInput` and `PyfuncOutput` part of the public MLflow API - i.e. do we expect users to import and interact with these directly?

If not, IMO, I think the `PyFuncInput` type hint is acceptable to drop in order to support a pandas/numpy-free skinny client since it's only used in an internal function. However, in accordance with https://github.com/mlflow/mlflow/pull/3972#pullrequestreview-568001973, it seems important to keep user-facing type hints around (e.g. `ModelSignature`, `Schema`, and `ModelInputExample` when used in user-facing MLflow Models functions).

Reply:
Doh, I missed an instance. This is a public APi.

Reply:
Left a comment in https://github.com/mlflow/mlflow/pull/3972/files#r557198770 indicating that these should be preserved as they're public-facing type hints.